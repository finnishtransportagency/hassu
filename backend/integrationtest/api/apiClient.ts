import { AbstractApi, OperationConfig } from "../../../common/abstractApi";
import { AppSyncResolverEvent } from "aws-lambda/trigger/appsync-resolver";
import { AppSyncEventArguments, handleEvent } from "../../src/apiHandler";

class API extends AbstractApi {
  async callAPI(operation: OperationConfig, variables?: unknown): Promise<unknown> {
    return (
      await handleEvent({
        info: { fieldName: operation.name },
        arguments: variables,
      } as AppSyncResolverEvent<AppSyncEventArguments>)
    ).data;
  }

  async callYllapitoAPI(operation: OperationConfig, variables?: unknown): Promise<unknown> {
    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
    const payload = {
      info: { fieldName: operation.name },
      arguments: variables,
    } as AppSyncResolverEvent<AppSyncEventArguments>;
    return (await handleEvent(payload)).data;
  }
}

export const api = new API();
