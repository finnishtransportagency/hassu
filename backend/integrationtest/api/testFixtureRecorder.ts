import fs from "fs";
import { projektiDatabase } from "../../src/database/projektiDatabase";
import { DBProjekti } from "../../src/database/model";
import { DocumentClient } from "aws-sdk/lib/dynamodb/document_client";
import { localDocumentClient } from "../util/databaseUtil";

export enum FixtureName {
  NAHTAVILLAOLO = "NAHTAVILLAOLO",
}

export async function recordProjektiTestFixture(fixtureName: string | FixtureName, oid: string): Promise<void> {
  const dbProjekti = await projektiDatabase.loadProjektiByOid(oid);
  replaceFieldsByName(dbProjekti, "2020-01-01T00:00:00+03:00", "tuotu", "paivitetty");
  replaceFieldsByName(dbProjekti, "salt123", "salt");

  const oldValue = readRecord(fixtureName);
  const currentValue = JSON.stringify(dbProjekti, null, 2);
  // Prevent updating file timestamp so that running tests with "watch" don't get into infinite loop
  if (!oldValue || oldValue !== currentValue) {
    fs.writeFileSync(__dirname + "/records/" + fixtureName + ".json", currentValue);
  }
}

function readRecord(fixtureName: string | FixtureName) {
  const buffer = fs.readFileSync(__dirname + "/records/" + fixtureName + ".json");
  return buffer.toString("utf-8");
}

export async function useProjektiTestFixture(fixtureName: string | FixtureName): Promise<string> {
  const dbProjekti: Partial<DBProjekti> = JSON.parse(readRecord(fixtureName));
  // Write the copied projekti to archive table
  const putParams: DocumentClient.PutItemInput = {
    TableName: process.env.TABLE_PROJEKTI,
    Item: dbProjekti,
  };
  await localDocumentClient.put(putParams).promise();

  return dbProjekti.oid;
}

export function replaceFieldsByName(obj: unknown, value: unknown, ...fieldNames: string[]): void {
  Object.keys(obj).forEach(function (prop) {
    if (typeof obj[prop] == "object" && obj[prop] !== null) {
      replaceFieldsByName(obj[prop], value, ...fieldNames);
    } else if (fieldNames.indexOf(prop) >= 0) {
      obj[prop] = value;
    }
  });
}
