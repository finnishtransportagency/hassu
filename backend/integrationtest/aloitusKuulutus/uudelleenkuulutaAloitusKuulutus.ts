import { api } from "../api/apiClient";
import assert from "assert";
import { TallennaProjektiInput, UudelleenKuulutusInput } from "hassu-common/graphql/apiModel";
import { tallennaEULogo } from "../api/testUtil/tests";

export async function uudelleenkuulutaAloitusKuulutus(oid: string, uudelleenKuulutusPaiva: string, saame?: boolean): Promise<void> {
  const projekti = await api.lataaProjekti(oid);
  assert(projekti.aloitusKuulutus?.uudelleenKuulutus);
  const uudelleenKuulutusInput: UudelleenKuulutusInput = {
    selosteKuulutukselle: {
      SUOMI: "Suomiseloste uudelleenkuulutukselle",
      RUOTSI: "Ruotsiseloste uudelleenkuulutukselle",
    },
    selosteLahetekirjeeseen: {
      SUOMI: "Suomiseloste uudelleenkuulutuksen lähetekirjeeseen",
      RUOTSI: "Ruotsiseloste uudelleenkuulutuksen lähetekirjeeseen",
    },
  };
  const { muokkausTila: _, aloituskuulutusSaamePDFt: _noNeedToSendThisFieldToAPI, ...rest } = projekti.aloitusKuulutus;
  const input: TallennaProjektiInput = {
    oid,
    versio: projekti.versio,
    aloitusKuulutus: {
      ...rest,
      kuulutusPaiva: uudelleenKuulutusPaiva,
      uudelleenKuulutus: uudelleenKuulutusInput,
    },
  };
  if (saame) {
    const uploadedIlmoitus = await tallennaEULogo("saameilmoitus.pdf");
    const uploadedKuulutus = await tallennaEULogo("saamekuulutus.pdf");
    input.aloitusKuulutus = {
      ...input.aloitusKuulutus,
      aloituskuulutusSaamePDFt: {
        POHJOISSAAME: { kuulutusPDFPath: uploadedKuulutus, kuulutusIlmoitusPDFPath: uploadedIlmoitus },
      },
    };
  }

  await api.tallennaProjekti(input);
}
