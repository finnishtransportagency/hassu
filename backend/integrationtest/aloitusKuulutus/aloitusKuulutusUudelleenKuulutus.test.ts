import { describe, it } from "mocha";
import * as sinon from "sinon";
import { KuulutusJulkaisuTila, Projekti, Status, TilasiirtymaToiminto, TilasiirtymaTyyppi } from "../../../common/graphql/apiModel";
import { UserFixture } from "../../test/fixture/userFixture";
import { userService } from "../../src/user";
import { aloitusKuulutusTilaManager } from "../../src/handler/tila/aloitusKuulutusTilaManager";
import { fileService } from "../../src/files/fileService";
import { FixtureName, useProjektiTestFixture } from "../api/testFixtureRecorder";
import { addLogoFilesToProjekti, defaultMocks, mockSaveProjektiToVelho } from "../api/testUtil/util";
import { deleteProjekti, testPublicAccessToProjekti, testYllapitoAccessToProjekti } from "../api/testUtil/tests";
import assert from "assert";
import { projektiDatabase } from "../../src/database/projektiDatabase";
import { assertIsDefined } from "../../src/util/assertions";
import { testProjektiDatabase } from "../../src/database/testProjektiDatabase";
import { uudelleenkuulutaAloitusKuulutus } from "./uudelleenkuulutaAloitusKuulutus";

const { expect } = require("chai");

describe("AloitusKuulutuksen uudelleenkuuluttaminen", () => {
  const userFixture = new UserFixture(userService);
  let publishProjektiFileStub: sinon.SinonStub;
  let oid: string;
  const { schedulerMock, emailClientStub, importAineistoMock, awsCloudfrontInvalidationStub, pdfGeneratorStub } = defaultMocks();

  before(async () => {
    publishProjektiFileStub = sinon.stub(fileService, "publishProjektiFile");
    publishProjektiFileStub.resolves();
    mockSaveProjektiToVelho();

    oid = await useProjektiTestFixture(FixtureName.ALOITUSKUULUTUS);
    try {
      await deleteProjekti(oid);
      awsCloudfrontInvalidationStub.reset();
    } catch (ignored) {
      // ignored
    }
    await addLogoFilesToProjekti(oid);
  });

  beforeEach(async () => {
    oid = await useProjektiTestFixture(FixtureName.ALOITUSKUULUTUS);
  });

  afterEach(() => {
    userFixture.logout();
    sinon.reset();
  });

  after(() => {
    sinon.restore();
  });

  it("should create uudelleenkuulutus for aloituskuulutus successfully", async function () {
    userFixture.loginAs(UserFixture.hassuAdmin);

    const initialProjekti = await projektiDatabase.loadProjektiByOid(oid);
    assertIsDefined(initialProjekti?.aloitusKuulutusJulkaisut?.[0]);
    // Aloituskuulutusjulkaisu on varmasti julkinen
    const kuulutusPaiva = "2000-01-01";
    const uudelleenKuulutusPaiva = "2031-01-01";
    initialProjekti.aloitusKuulutusJulkaisut[0].kuulutusPaiva = kuulutusPaiva;
    initialProjekti.aloitusKuulutusJulkaisut[0].siirtyySuunnitteluVaiheeseen = "2222-01-01";
    await testProjektiDatabase.saveProjekti({
      oid,
      aloitusKuulutusJulkaisut: initialProjekti.aloitusKuulutusJulkaisut,
    });

    // Aktivoi uudelleenkuulutus julkaistulle aloituskuulutukselle
    await aloitusKuulutusTilaManager.siirraTila({
      oid,
      toiminto: TilasiirtymaToiminto.UUDELLEENKUULUTA,
      tyyppi: TilasiirtymaTyyppi.ALOITUSKUULUTUS,
    });
    await testYllapitoAccessToProjekti(oid, Status.SUUNNITTELU, " aloituskuulutus uudelleenkuulutus avattu", (projekti) => {
      const { aloitusKuulutus } = projekti;
      return { uudelleenKuulutus: aloitusKuulutus?.uudelleenKuulutus };
    });

    // Lisätään uudelleenkuulutukseen selitystekstit
    await uudelleenkuulutaAloitusKuulutus(oid, uudelleenKuulutusPaiva);
    await testYllapitoAccessToProjekti(
      oid,
      Status.SUUNNITTELU,
      " aloituskuulutus uudelleenkuulutuksen selitystekstit täytetty",
      (projekti: Projekti) => {
        const { aloitusKuulutus } = projekti;
        return { uudelleenKuulutus: aloitusKuulutus?.uudelleenKuulutus };
      }
    );

    // Hyväksytään uudelleenkuulutus
    userFixture.loginAs(UserFixture.mattiMeikalainen);
    await aloitusKuulutusTilaManager.siirraTila({
      oid,
      toiminto: TilasiirtymaToiminto.LAHETA_HYVAKSYTTAVAKSI,
      tyyppi: TilasiirtymaTyyppi.ALOITUSKUULUTUS,
    });

    userFixture.loginAs(UserFixture.projari112);
    await aloitusKuulutusTilaManager.siirraTila({
      oid,
      toiminto: TilasiirtymaToiminto.HYVAKSY,
      tyyppi: TilasiirtymaTyyppi.ALOITUSKUULUTUS,
    });

    await testYllapitoAccessToProjekti(oid, Status.SUUNNITTELU, "aloituskuulutus uudelleenkuulutus hyväksytty", (projekti: Projekti) => {
      const { aloitusKuulutus, aloitusKuulutusJulkaisu } = projekti;
      return {
        aloitusKuulutusJulkaisu: { tila: aloitusKuulutusJulkaisu?.tila, uudelleenKuulutus: aloitusKuulutusJulkaisu?.uudelleenKuulutus },
        uudelleenKuulutus: aloitusKuulutus?.uudelleenKuulutus,
      };
    });

    const resultProjekti = await projektiDatabase.loadProjektiByOid(oid);
    assert(resultProjekti);
    assert(resultProjekti.aloitusKuulutusJulkaisut);
    expect(resultProjekti.aloitusKuulutusJulkaisut).to.have.length(2);

    const alkuperainenJulkaisu = resultProjekti.aloitusKuulutusJulkaisut[0];
    expect(alkuperainenJulkaisu.id).to.eq(1);
    expect(alkuperainenJulkaisu.tila).to.eq(KuulutusJulkaisuTila.HYVAKSYTTY);
    expect(alkuperainenJulkaisu.kuulutusPaiva).to.eq(kuulutusPaiva);

    const uudelleenKuulutusJulkaisu = resultProjekti.aloitusKuulutusJulkaisut[1];
    expect(uudelleenKuulutusJulkaisu.id).to.eq(2);
    expect(uudelleenKuulutusJulkaisu.tila).to.eq(KuulutusJulkaisuTila.HYVAKSYTTY);
    expect(uudelleenKuulutusJulkaisu.kuulutusPaiva).to.eq(uudelleenKuulutusPaiva);

    emailClientStub.verifyEmailsSent();
    pdfGeneratorStub.verifyAllPDFContents();
    await importAineistoMock.processQueue();

    uudelleenKuulutusJulkaisu.kuulutusPaiva = "2000-01-01"; // Simuloidaan ajan kulumista asettamalla kuulutuspäivä varmasti menneisyyteen, jotta julkaisu on julkinen
    await testProjektiDatabase.aloitusKuulutusJulkaisut.update(resultProjekti, uudelleenKuulutusJulkaisu);
    await schedulerMock.verifyAndRunSchedule();
    awsCloudfrontInvalidationStub.verifyCloudfrontWasInvalidated();

    await testPublicAccessToProjekti(oid, Status.ALOITUSKUULUTUS, userFixture, " uudelleenkuulutuksen jälkeen", (julkinen) => {
      return julkinen.aloitusKuulutusJulkaisu;
    });
  });
});
