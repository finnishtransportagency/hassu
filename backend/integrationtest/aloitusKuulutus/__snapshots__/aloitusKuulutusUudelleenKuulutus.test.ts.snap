// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AloitusKuulutuksen uudelleenkuuluttaminen should create uudelleenkuulutus for aloituskuulutus successfully 1`] = `
Object {
  "description": "yllapitoProjekti aloituskuulutus uudelleenkuulutus avattu",
  "mockedtime": "2020-01-01T02:00",
  "obj": Object {
    "uudelleenKuulutus": Object {
      "__typename": "UudelleenKuulutus",
      "alkuperainenHyvaksymisPaiva": "2022-01-02T00:00:01+02:00",
      "alkuperainenKuulutusPaiva": "2000-01-01",
      "selosteKuulutukselle": undefined,
      "selosteLahetekirjeeseen": undefined,
      "tiedotaKiinteistonomistajia": undefined,
      "tila": "JULKAISTU_PERUUTETTU",
    },
  },
}
`;

exports[`AloitusKuulutuksen uudelleenkuuluttaminen should create uudelleenkuulutus for aloituskuulutus successfully 2`] = `
Object {
  "description": "yllapitoProjekti aloituskuulutus uudelleenkuulutuksen selitystekstit täytetty",
  "mockedtime": "2020-01-01T02:00",
  "obj": Object {
    "uudelleenKuulutus": Object {
      "__typename": "UudelleenKuulutus",
      "alkuperainenHyvaksymisPaiva": "2022-01-02T00:00:01+02:00",
      "alkuperainenKuulutusPaiva": "2000-01-01",
      "selosteKuulutukselle": Object {
        "RUOTSI": "Ruotsiseloste uudelleenkuulutukselle",
        "SUOMI": "Suomiseloste uudelleenkuulutukselle",
        "__typename": "LokalisoituTeksti",
      },
      "selosteLahetekirjeeseen": Object {
        "RUOTSI": "Ruotsiseloste uudelleenkuulutuksen lähetekirjeeseen",
        "SUOMI": "Suomiseloste uudelleenkuulutuksen lähetekirjeeseen",
        "__typename": "LokalisoituTeksti",
      },
      "tiedotaKiinteistonomistajia": undefined,
      "tila": "JULKAISTU_PERUUTETTU",
    },
  },
}
`;

exports[`AloitusKuulutuksen uudelleenkuuluttaminen should create uudelleenkuulutus for aloituskuulutus successfully 3`] = `
Object {
  "description": "yllapitoProjektialoituskuulutus uudelleenkuulutus hyväksytty",
  "mockedtime": "2020-01-01T02:00",
  "obj": Object {
    "aloitusKuulutusJulkaisu": Object {
      "tila": "HYVAKSYTTY",
      "uudelleenKuulutus": Object {
        "__typename": "UudelleenKuulutus",
        "alkuperainenHyvaksymisPaiva": "2022-01-02T00:00:01+02:00",
        "alkuperainenKuulutusPaiva": "2000-01-01",
        "selosteKuulutukselle": Object {
          "RUOTSI": "Ruotsiseloste uudelleenkuulutukselle",
          "SUOMI": "Suomiseloste uudelleenkuulutukselle",
          "__typename": "LokalisoituTeksti",
        },
        "selosteLahetekirjeeseen": Object {
          "RUOTSI": "Ruotsiseloste uudelleenkuulutuksen lähetekirjeeseen",
          "SUOMI": "Suomiseloste uudelleenkuulutuksen lähetekirjeeseen",
          "__typename": "LokalisoituTeksti",
        },
        "tiedotaKiinteistonomistajia": undefined,
        "tila": "JULKAISTU_PERUUTETTU",
      },
    },
    "uudelleenKuulutus": null,
  },
}
`;

exports[`AloitusKuulutuksen uudelleenkuuluttaminen should create uudelleenkuulutus for aloituskuulutus successfully 4`] = `
Array [
  Object {
    "subject": "Valtion liikenneväylien suunnittelu: Aloituskuulutus odottaa hyväksyntää HASSU/123/2023",
    "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektistasi HASSU AUTOMAATTITESTIPROJEKTI1 on luotu aloituskuulutus, joka odottaa hyväksyntääsi.

Voit tarkastella kuulutusta osoitteessa https://localhost:3000/yllapito/projekti/1.2.246.578.5.1.2978288874.2711575506/aloituskuulutus

Sait tämän viestin, koska sinut on merkitty projektin projektipäälliköksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
    "to": Array [
      "mikko.haapamki@cgi.com",
      "mikko.haapamaki02@cgi.com",
    ],
  },
  Object {
    "subject": "Valtion liikenneväylien suunnittelu: Aloituskuulutus hyväksytty HASSU/123/2023",
    "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektisi HASSU AUTOMAATTITESTIPROJEKTI1 aloituskuulutus on hyväksytty.

Voit tarkastella kuulutusta osoitteessa https://localhost:3000/yllapito/projekti/1.2.246.578.5.1.2978288874.2711575506/aloituskuulutus

Sait tämän viestin, koska sinut on merkitty kuulutuksen laatijaksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
    "to": "mikko.haapamaki@cgi.com",
  },
  Object {
    "attachments": Array [
      Object {
        "contentType": "application/pdf",
        "filename": "T412 Aloituskuulutus.pdf",
      },
    ],
    "subject": "Valtion liikenneväylien suunnittelu: Aloituskuulutus hyväksytty HASSU/123/2023",
    "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektisi HASSU AUTOMAATTITESTIPROJEKTI1 aloituskuulutus on hyväksytty.

Voit tarkastella kuulutusta osoitteessa https://localhost:3000/yllapito/projekti/1.2.246.578.5.1.2978288874.2711575506/aloituskuulutus

Viethän sekä oheisen kuulutuksen että erillisen viestin, jossa on liitteenä ilmoitus kuulutuksesta, asianhallintaan suunnitelman hallinnollisen käsittelyn asialle. Toimi organisaatiosi asianhallinnan ohjeistusten mukaisesti.

Sait tämän viestin, koska sinut on merkitty projektin projektipäälliköksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
    "to": Array [
      "mikko.haapamki@cgi.com",
      "mikko.haapamaki02@cgi.com",
    ],
  },
  Object {
    "attachments": Array [
      Object {
        "contentType": "application/pdf",
        "filename": "T412_1 Ilmoitus aloituskuulutuksesta.pdf",
      },
    ],
    "cc": Array [
      "mikko.haapamki@cgi.com",
      "mikko.haapamaki02@cgi.com",
    ],
    "subject": "Ilmoitus Väyläviraston kuulutuksesta",
    "text": "Tämä viesti on lähetetty Valtion liikenneväylien suunnittelu -järjestelmästä, älä vastaa tähän viestiin.

HASSU AUTOMAATTITESTIPROJEKTI1

Suomiseloste uudelleenkuulutuksen lähetekirjeeseen

Nokia ilmoittaa, että se julkaisee kuulutuksen, joka koskee otsikossa mainitun tiesuunnitelman ja maastotöiden aloittamista. Nokia saattaa asian tiedoksi julkisesti kuuluttamalla vähintään 30 päivän ajan siten, kuin julkisesta kuulutuksesta säädetään hallintolaissa, sekä julkaisemalla kuulutuksen vähintään yhdessä alueella yleisesti ilmestyvässä sanomalehdessä. (laki liikennejärjestelmästä ja maanteistä LjMTL 108§, hallintolaki HL 62 a §)

Kuulutus julkaistaan 1.1.2031 osoitteessa https://localhost:3000/s/3c1d/aloituskuulutus.

Suunnittelualueen kuntia pyydetään julkaisemaan liitteenä oleva ilmoitus kuulutuksesta verkkosivuillaan. Ilmoitus tulee julkaista mahdollisuuksien mukaan edellä mainittuna kuulutuksen julkaisupäivänä. Ilmoituksen nähtävillä oloa ei tarvitse todentaa ilmoituksen lähettäjälle. Ilmoitus menee ELY-keskuksen ja Väyläviraston kuulutussivulle automaattisesti Valtion liikenneväylien suunnittelu palvelun kautta.

Lorem Ipsum

Väylävirasto käsittelee suunnitelman laatimiseen liittyen tarpeellisia henkilötietoja. Halutessasi tietää tarkemmin väyläsuunnittelun tietosuojakäytänteistä tutustu verkkosivujen tietosuojaosioon, https://www.vayla.fi/tietosuoja.

Lisätietoja antaa 

Nokia, Testi1 Hassu, puh. 029123123, mikko.haapamaki1@cgi.com

Kajaani, Marko Koi, puh. 0293121213, markku.koi@koi.com

",
    "to": Array [
      "mikkeli@mikke.li",
      "juva@ju.va",
      "savonlinna@savonlin.na",
      "kirjaamo.etela-savo@ely-keskus.fi",
    ],
  },
]
`;

exports[`AloitusKuulutuksen uudelleenkuuluttaminen should create uudelleenkuulutus for aloituskuulutus successfully 5`] = `
Object {
  "description": "PDF:ien sisällöt",
  "mockedtime": "2020-01-01T02:00",
  "obj": Array [
    Object {
      "nimi": "T412 Aloituskuulutus.pdf",
      "textContent": "HASSU/123/2023



Kuulutus suunnittelun aloittamisesta
HASSU AUTOMAATTITESTIPROJEKTI1
Suomiseloste uudelleenkuulutukselle
Nokia, sovittuaan asiasta Väyläviraston kanssa, aloittaa otsikon mukaisen tiesuunnitelman laatimisen ja suunnitelman vaatimat maastotyöt ja tutkimukset.
Lorem Ipsum
Nokia on 1.1.2031 julkaissut kuulutuksen suunnittelun aloittamisesta ja maastotutkimuksista osoitteessa https://localhost:3000/s/3c1d/aloituskuulutus. Kuulutus pidetään nähtävillä 1.1.2031 - 31.1.2031.

Asianosaisten katsotaan saaneen tiedon suunnittelun käynnistymisestä ja tutkimusoikeudesta seitsemäntenä päivänä kuulutuksen julkaisusta (hallintolaki 62 a §). Tämän jälkeen suunnitelmasta vastaavalla on oikeus tehdä kiinteistöillä suunnittelutyön vaatimia mittauksia, maaperätutkimuksia ja muita valmistelevia toimenpiteitä. Asianosaisilla on oikeus olla tutkimuksissa paikalla ja ilmaista mielipiteensä asiassa. (laki liikennejärjestelmästä ja maanteistä LjMTL 16 §).
Suunnitelman luonnosvaiheessa tiedotetaan yleisesti mahdollisuudesta kertoa mielipide suunnitelmasta. Kun suunnitelma on valmis, se asetetaan yleisesti nähtäville. Nähtäville asettamisesta julkaistaan kuulutus ja tiedotetaan asianosaisia. Tässä vaiheessa suunnitelmasta on mahdollisuus tehdä kirjallinen muistutus. (LjMTL 27 § ja valtioneuvoston asetus maanteistä 3 §). 
Väylävirasto käsittelee suunnitelman laatimiseen liittyen tarpeellisia henkilötietoja. Halutessasi tietää tarkemmin väyläsuunnittelun tietosuojakäytänteistä tutustu verkkosivujen tietosuojaosioon, https://www.vayla.fi/tietosuoja.

Lisätietoja antaa 
Testi1 Hassu (Nokia), 
puh. 029123123, mikko.haapamaki1(at)cgi.com 
Marko Koi (Kajaani), 
puh. 0293121213, markku.koi(at)koi.com 

",
    },
    Object {
      "nimi": "T412_1 Ilmoitus aloituskuulutuksesta.pdf",
      "textContent": "HASSU/123/2023



Ilmoitus aloituskuulutuksesta
HASSU AUTOMAATTITESTIPROJEKTI1
Suomiseloste uudelleenkuulutukselle
Nokia, sovittuaan asiasta Väyläviraston kanssa julkaisee kuulutuksen, joka koskee otsikossa mainitun tiesuunnitelman suunnittelun ja maastotöiden aloittamista. Nokia saattaa asian tiedoksi julkisesti kuuluttamalla siten, kuin julkisesta kuulutuksesta säädetään hallintolaissa, sekä julkaisemalla kuulutuksen vähintään yhdessä alueella yleisesti ilmestyvässä sanomalehdessä. (laki liikennejärjestelmästä ja maanteistä LjMTL 108§, hallintolaki HL 62 a §)
Kuulutus julkaistaan 1.1.2031 osoitteessa https://localhost:3000/s/3c1d/aloituskuulutus



Kuulutus on nähtävillä 1.1.2031 - 31.1.2031 välisen ajan
Lisätietoja antaa 
Testi1 Hassu (Nokia), 
puh. 029123123, mikko.haapamaki1(at)cgi.com 
Marko Koi (Kajaani), 
puh. 0293121213, markku.koi(at)koi.com 

",
    },
  ],
}
`;

exports[`AloitusKuulutuksen uudelleenkuuluttaminen should create uudelleenkuulutus for aloituskuulutus successfully 6`] = `Array []`;

exports[`AloitusKuulutuksen uudelleenkuuluttaminen should create uudelleenkuulutus for aloituskuulutus successfully 7`] = `
Object {
  "args": Object {
    "DistributionId": "unit-test-distribution-id",
    "InvalidationBatch": Object {
      "CallerReference": "***unittest***",
      "Paths": Object {
        "Items": Array [
          "/tiedostot/suunnitelma/1.2.246.578.5.1.2978288874.2711575506/*",
        ],
        "Quantity": 1,
      },
    },
  },
  "stub": "createInvalidation",
}
`;

exports[`AloitusKuulutuksen uudelleenkuuluttaminen should create uudelleenkuulutus for aloituskuulutus successfully 8`] = `
Object {
  "description": "publicProjekti  uudelleenkuulutuksen jälkeen",
  "mockedtime": "2020-01-01T02:00",
  "obj": Object {
    "__typename": "AloitusKuulutusJulkaisuJulkinen",
    "aloituskuulutusSaamePDFt": undefined,
    "hankkeenKuvaus": Object {
      "RUOTSI": "På Svenska",
      "SUOMI": "Lorem Ipsum",
      "__typename": "LokalisoituTeksti",
    },
    "id": 2,
    "julkaisuOnKopio": false,
    "kielitiedot": Object {
      "__typename": "Kielitiedot",
      "ensisijainenKieli": "SUOMI",
    },
    "kuulutusPDF": Object {
      "SUOMI": "/tiedostot/suunnitelma/1.2.246.578.5.1.2978288874.2711575506/aloituskuulutus/T412 Aloituskuulutus.pdf",
      "__typename": "KuulutusPDFJulkinen",
    },
    "kuulutusPaiva": "2000-01-01",
    "kuulutusTekstit": Object {
      "__typename": "KuulutusTekstit",
      "infoTekstit": Array [
        "Asianosaisten katsotaan saaneen tiedon suunnittelun käynnistymisestä ja tutkimusoikeudesta seitsemäntenä päivänä kuulutuksen julkaisusta (hallintolaki 62 a §). Tämän jälkeen suunnitelmasta vastaavalla on oikeus tehdä kiinteistöillä suunnittelutyön vaatimia mittauksia, maaperätutkimuksia ja muita valmistelevia toimenpiteitä. Asianosaisilla on oikeus olla tutkimuksissa paikalla ja ilmaista mielipiteensä asiassa. (laki liikennejärjestelmästä ja maanteistä LjMTL 16 §).",
        "Suunnitelman luonnosvaiheessa tiedotetaan yleisesti mahdollisuudesta kertoa mielipide suunnitelmasta. Kun suunnitelma on valmis, se asetetaan yleisesti nähtäville. Nähtäville asettamisesta julkaistaan kuulutus ja tiedotetaan asianosaisia. Tässä vaiheessa suunnitelmasta on mahdollisuus tehdä kirjallinen muistutus. (LjMTL 27 § ja valtioneuvoston asetus maanteistä 3 §). ",
      ],
      "kuvausTekstit": Array [
        "Nokia on 1.1.2000 julkaissut kuulutuksen suunnittelun aloittamisesta ja maastotutkimuksista osoitteessa <a href=\\"https://localhost:3000/s/3c1d/aloituskuulutus\\">https://localhost:3000/s/3c1d/aloituskuulutus</a>. Kuulutus pidetään nähtävillä <strong>1.1.2000 - 31.1.2000</strong>.",
      ],
      "leipaTekstit": Array [
        "Nokia, sovittuaan asiasta Väyläviraston kanssa, aloittaa otsikon mukaisen tiesuunnitelman laatimisen ja suunnitelman vaatimat maastotyöt ja tutkimukset.",
      ],
      "tietosuoja": "Väylävirasto käsittelee suunnitelman laatimiseen liittyen tarpeellisia henkilötietoja. Halutessasi tietää tarkemmin väyläsuunnittelun tietosuojakäytänteistä tutustu verkkosivujen tietosuojaosioon, <a external=\\"true\\" href=\\"https://www.vayla.fi/tietosuoja\\">https://www.vayla.fi/tietosuoja</a>.",
    },
    "siirtyySuunnitteluVaiheeseen": "2023-01-01",
    "suunnitelmaJaettu": undefined,
    "suunnitteluSopimus": Object {
      "__typename": "SuunnitteluSopimusJulkaisu",
      "email": "mikko.haapamaki1@cgi.com",
      "etunimi": "Testi1",
      "kunta": 536,
      "logo": Object {
        "RUOTSI": undefined,
        "SUOMI": "/tiedostot/suunnitelma/1.2.246.578.5.1.2978288874.2711575506/suunnittelusopimus/logo.png",
        "__typename": "LokalisoituTeksti",
      },
      "puhelinnumero": "029123123",
      "sukunimi": "Hassu",
    },
    "tila": "HYVAKSYTTY",
    "uudelleenKuulutus": Object {
      "__typename": "UudelleenKuulutus",
      "alkuperainenHyvaksymisPaiva": "2022-01-02T00:00:01+02:00",
      "selosteKuulutukselle": Object {
        "RUOTSI": "Ruotsiseloste uudelleenkuulutukselle",
        "SUOMI": "Suomiseloste uudelleenkuulutukselle",
        "__typename": "LokalisoituTeksti",
      },
      "selosteLahetekirjeeseen": null,
      "tila": "JULKAISTU_PERUUTETTU",
    },
    "velho": Object {
      "__typename": "VelhoJulkinen",
      "asiatunnusELY": undefined,
      "asiatunnusVayla": "HASSU/123/2023",
      "geoJSON": undefined,
      "kunnat": Array [
        91,
        92,
      ],
      "kuvaus": undefined,
      "linkki": "https://linkki-hankesivulle.fi",
      "maakunnat": Array [
        1,
      ],
      "nimi": "HASSU AUTOMAATTITESTIPROJEKTI1",
      "suunnittelustaVastaavaViranomainen": "VAYLAVIRASTO",
      "toteuttavaOrganisaatio": undefined,
      "tyyppi": "TIE",
      "vaylamuoto": Array [
        "tie",
      ],
    },
    "yhteystiedot": Array [
      Object {
        "__typename": "Yhteystieto",
        "etunimi": "Testi1",
        "kunta": 536,
        "organisaatio": "CGI Suomi Oy",
        "puhelinnumero": "029123123",
        "sahkoposti": "mikko.haapamaki1@cgi.com",
        "sukunimi": "Hassu",
      },
      Object {
        "__typename": "Yhteystieto",
        "etunimi": "Marko",
        "organisaatio": "Kajaani",
        "puhelinnumero": "0293121213",
        "sahkoposti": "markku.koi@koi.com",
        "sukunimi": "Koi",
      },
    ],
  },
}
`;
